internal/pkg/manifest/validate.go

```diff
@@ -2067,9 +2067,9 @@ func contains(name string, names []string) bool {
 	return false
 }
 
-// validate returns nil if SidecarImageConfig is configured correctly.
-func (cfg SidecarImageConfig) validate() error {
-	if !cfg.Build.IsZero() && cfg.Location != nil {
+// validate returns nil if ImageLocationOrBuild is configured correctly.
+func (i ImageLocationOrBuild) validate() error {
```

Comment:
Is this function invoked for the main container?

Reply:
No, this is just to validate sidecar containers

Reply:
yeah but main container also includes `ImageLocationOrBuild`. Shall we invoke it in `Image.validate()` to make sure users specifying the correct `image` section in their manifests?

Reply:
yup adding to the `Image` section as well

Reply:
Both cases are handled now well in validation
1. Both `Build` and `location` are empty
2. Both  `Build` and `location` are not empty